claat/proto-renderer/generic_renderer_test.go

```diff
@@ -54,7 +35,26 @@ func TestExecuteTemplate(t *testing.T) {
 		{newSampleProtoTemplate("not-valid"), "", true},
 	}
 
-	for _, test := range tests {
-		runEncapsulatedTest(test, tmpl, t)
+	for _, tc := range tests {
+		runEncapsulatedTest(tc, tmpl, t)
+	}
+}
+
+func runEncapsulatedTest(test encapsulatedTest, tmpl *template.Template, t *testing.T) {
+	// Check wheather template failed to render by checking for panic
+	defer func(test encapsulatedTest) {
+		err := recover()
+		if err != nil && test.ok {
+			t.Errorf("\nFor:\n\t%#v\nPanic occured:\n\t%s\t(false negative)", test, err)
+		}
+
+		if err == nil && !test.ok {
+			t.Errorf("\nPanic did not occur for:\n\t%#v\n\t(false positive)", test)
+		}
+	}(test)
+
+	tmplOut := ExecuteTemplate(test.in, tmpl)
+	if test.out != tmplOut {
+		t.Errorf("Expecting:\n\t'%s'\nBut got:\n\t'%s'", test.out, tmplOut)
```

Comment:
Try output format like:

ExecuteTemplate(%v, %v) = %s, want %s

It notes the input args and prominently names the problematic function as well.

Reply:
> names the problematic function as well

I thought the test function name (`TestExecuteTemplate`) already defines that, but ok. Right now outputs look as following:

![image](https://user-images.githubusercontent.com/4205668/59795059-60efcc80-92a8-11e9-968b-b570285e7171.png)

having `%#v` vs plain `%v` shows the inner type values, which can be helpful

Reply:
Now:
![image](https://user-images.githubusercontent.com/4205668/59795526-6ef21d00-92a9-11e9-8cc7-9177c2b34a4c.png)